dandi/organize.py

```diff
@@ -857,7 +861,7 @@ def _get_metadata(path):
         act(os.makedirs, dandiset_path)
 
     if files_mode == "auto":
-        files_mode = detect_link_type(dandiset_path)
+        files_mode = detect_link_type(srcdir, dandiset_path)
```

Comment:
why do we have to take directory here for `srcdir` and can't just take a `metadata[0]['path']` as the source file to test linking against as I thought about in https://github.com/dandi/dandi-cli/issues/1110#issuecomment-1244429099 ?

Reply:
@yarikoptic Before you wrote that comment, I had already modified `detect_link_type()` to take two directory paths.  I don't see an advantage to changing it to take a file path.

Reply:
Re advantage: User might not have write permissions to modify the `srcdir` - e.g. it belongs to read-only lab account.  Current `master` code would work ok since we create `srcdir` also in the target dandiset, so for such use cases it would work.  With the change you propose code would start to error out since user would not be able to create anything in `srcdir`, so it would be a regression we better avoid introducing.